linear-base.cabal

```
@@ -171,38 +189,28 @@ test-suite test
         tasty,
         tasty-hedgehog >= 1.2,
         mmorph,
-        vector
+        vector,
+        linear-generics
 
-test-suite examples
+test-suite test-examples
```

```suggestion
test-suite examples
```
Should work. Because different units are namespaced between their types, so it'll be referred to as something like `test:examples` (instead of `test:test-examples` in your case).

---

It doesnt seem to work:

```
library examples
    import: build-opts
    hs-source-dirs: examples
    
[...]

test-suite examples
    import: build-opts
    import: rts-opts-multithread
    type:             exitcode-stdio-1.0
    main-is:          Main.hs
    hs-source-dirs:   test/examples
```

gives

```
Configuring linear-base-0.4.0...
There exist several components with the same name: 'examples'. This is a bug in your .cabal file.
CallStack (from HasCallStack):
  error, called at libraries/Cabal/Cabal-syntax/src/Distribution/PackageDescription/Utils.hs:18:15 in Cabal-syntax-3.10.3.0:Distribution.PackageDescription.Utils
Progress 1/2

Error: [S-7282]
       Stack failed to execute the build plan.
       
       While executing the build plan, Stack encountered the error:
       
       [S-7011]
       While building package linear-base-0.4.0 (scroll up to its section to see the error)
       using:
[...]
```

---

My bad then.